kale/loaddata/few_shot.py

```diff
@@ -0,0 +1,107 @@
+import os
+from typing import Any
+
+import numpy as np
+import torch
+from PIL import Image
+from torch.utils.data import Dataset
+
+
+class NWayKShotDataset(Dataset):
+    """
+    It is used to load data for N-way K-shot learning using Prototypical Networks.
+
+    Note:
+        The dataset should be organized as:
+
+        - root
+            - train
+                - class_name 1
+                    - xxx.png
+                    - yyy.png
+                    - ...
+                - class_name 2
+                    - xxx.png
+                    - yyy.png
+                    - ...
+                - ...
+            - val
+                - class_name 1
+                    - xxx.png
+                    - yyy.png
+                    - ...
+                - class_name 2
+                    - xxx.png
+                    - yyy.png
+                    - ...
+                - ...
+            - test (optional)
+                - class_name 1
+                    - xxx.png
+                    - yyy.png
+                    - ...
+                - class_name 2
+                    - xxx.png
+                    - yyy.png
+                    - ...
+                - ...
+
+    Args:
+        path (string): The root directory of the data.
+        mode (string): The mode of the dataset. It can be 'train', 'val' or 'test'. Default: 'train'.
```

Comment:
Double-check whether we call this the `mode` elsewhere, or something else.

Reply:
Some of them used `train=True`. But here I need 'train', 'val' and 'test' to index data. Just using flag indicating train or not is not enough.

Reply:
See what ChatGPT says (sent via Skype Chat to keep the formatting).

Reply:
Unless you can find a reputable source using "mode" for this purpose (in the PyTorch ecosystem if possible), I lean towards changing it.

Reply:
A quick example is the MedKLIP baseline I am using now. It used `mode='train'` in `model_MedKLIP.MedKLIP.__init__()`

Reply:
This still needs to be reviewed for better English words. MedKLIP's English is not their strength. Find something in PyTorch or better software packages. See my comments on Jan 4. I do not think this is "mode". We need to resolve this in May. Discuss with Shuo/Xianyuan if needed.

Reply:
There are quite a lot of repos that use `mode`. For example, [Pytorch-GAN](https://github.com/search?q=repo%3Aeriklindernoren%2FPyTorch-GAN%20mode%3D%27train%27&type=code) with 15.7k stars, Baidu's [paddlepaddle](https://github.com/search?q=repo%3APaddlePaddle%2FPaddleSeg%20mode%3D%27train&type=code) and [ultralytics yolov8](https://github.com/ultralytics/ultralytics/blob/b6415e4ef99a0a6447db85450dab2802df78da7a/ultralytics/cfg/default.yaml#L29) with 23.2k stars.